bmaptools/TransRead.py

```diff
@@ -499,6 +499,7 @@ def _print_warning(timeout):
         else:
             import http.client as httplib
             import urllib.request as urllib2
+            self._force_fake_seek = False
```

Comment:
I don't understand why this is desirable. `_force_fake_seek = True` emulates seeking in terms of reading and discarding the unwanted data. Your second traceback seems like a consequence of needing `_force_fake_seek` but not getting it? I don't see how explicitly disabling it would help you?

Perhaps something like this would resolve the second traceback?

```
try:
    self._f_objs[-1].seek(offset, whence)
except io.UnsupportedOperation:
    self._force_fake_seek = True
    self.seek(offset, whence)
````

Reply:
The more fundamental issue is that bmap-tools currently assumes there are two possibilities for whether a file-like object is seekable:

* yes it's seekable, and seek() works
* no it isn't seekable, and there is no seek attribute

but in fact there is a third:

* no it isn't seekable, but we only find this out at runtime when seek() raises UnsupportedOperation

Setting `_force_fake_seek = True` for the `Popen` code paths is a workaround for not dealing with that third case, or an optimization for already knowing ahead of time that we are in the third case, depending how you look at it.

I'm surprised that installing directly from a URL doesn't also need `_force_fake_seek = True`. (That isn't a feature that I routinely use, so I'm not surprised I didn't see that bug before.)